### PR TITLE
Fix cross join with parallel replicas.

### PR DESCRIPTION
**Pull request by KochetovNicolai**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Fixing `Unexpected node type for table expression` after https://github.com/ClickHouse/ClickHouse/pull/72856

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

#### CI Settings (Only check the boxes if you know what you are doing):
- [ ]  Allow: All Required Checks
- [ ]  Allow: Stateless tests
- [ ]  Allow: Stateful tests
- [ ]  Allow: Integration Tests
- [ ]  Allow: Performance tests
- [ ]  Allow: All Builds
- [ ]  Allow: batch 1, 2 for multi-batch jobs
- [ ]  Allow: batch 3, 4, 5, 6 for multi-batch jobs
---
- [ ]  Exclude: Style check
- [ ]  Exclude: Fast test
- [ ]  Exclude: All with ASAN
- [ ]  Exclude: All with TSAN, MSAN, UBSAN, Coverage
- [ ]  Exclude: All with aarch64, release, debug
---
- [ ]  Run only fuzzers related jobs (libFuzzer fuzzers, AST fuzzers, etc.)
- [ ]  Exclude: AST fuzzers
---
- [ ]  Do not test
- [ ]  Woolen Wolfdog
- [ ]  Upload binaries for special builds
- [ ]  Disable merge-commit
- [ ]  Disable CI cache

### CODE REVIEW

**Comment by robot-clickhouse-ci-2** [comment]
*This is an automated comment for commit dd64aed30058acba4fca0e77994e36fcef482773 with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/73451/dd64aed30058acba4fca0e77994e36fcef482773/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Docs check</td><td>Builds and tests the documentation</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/a91d8a6dbfbae001526cbf42e019f46efa59b5bd/docs_check.html">✅ success</a></td></tr>
<tbody>
</table>
</details>

File: tests/queries/0_stateless/03286_parallel_replicas_cross_join_bug.sql [hunk 1]
```diff
@@ -0,0 +1,9 @@
+drop table if exists tab;
+create table tab (x UInt64) engine = MergeTree order by tuple();
+insert into tab select number from numbers(1e7);
+
+set allow_experimental_parallel_reading_from_replicas = 1, max_parallel_replicas = 3, cluster_for_parallel_replicas = 'parallel_replicas', parallel_replicas_for_non_replicated_merge_tree = true, parallel_replicas_local_plan = false;
```
Comment (devcrafter) [inline tests/queries/0_stateless/03286_parallel_replicas_cross_join_bug.sql:5]:
```suggestion
set enable_parallel_replicas = 1, max_parallel_replicas = 3, cluster_for_parallel_replicas = 'parallel_replicas', parallel_replicas_for_non_replicated_merge_tree = true, parallel_replicas_local_plan = false;
```

**Review by devcrafter — COMMENTED** [review]
Why disable local plan (the setting is randomized, btw)?

File: tests/queries/0_stateless/03286_parallel_replicas_cross_join_bug.sql [hunk 2]
```diff
@@ -0,0 +1,9 @@
+drop table if exists tab;
+create table tab (x UInt64) engine = MergeTree order by tuple();
+insert into tab select number from numbers(1e7);
+
+set enable_parallel_replicas = 1, max_parallel_replicas = 3, cluster_for_parallel_replicas = 'parallel_replicas', parallel_replicas_for_non_replicated_merge_tree = true, parallel_replicas_local_plan = false;
```
Comment (devcrafter) [inline tests/queries/0_stateless/03286_parallel_replicas_cross_join_bug.sql:5]: Why disable local plan (the setting is randomized, btw)?

Comment (KochetovNicolai) [reply tests/queries/0_stateless/03286_parallel_replicas_cross_join_bug.sql:5]: That was just a subset of settings that did reproduce the issue. Let me remove the one with local plan.

**Review by devcrafter — APPROVED** [review]

File: src/Planner/findParallelReplicasQuery.cpp [hunk 3]
```diff
@@ -401,6 +401,11 @@ static const TableNode * findTableForParallelReplicas(const IQueryTreeNode * que
                 query_tree_node = array_join_node.getTableExpression().get();
                 break;
             }
+            case QueryTreeNodeType::CROSS_JOIN:
+            {
+                /// TODO: We can parallelize one table
```
Comment (devcrafter) [inline src/Planner/findParallelReplicasQuery.cpp:406]: Forgot, - let's add test with CROSS JOIN in subquery with PR to cover this lines

**Pull request merged**